test/python/circuit/test_controlled_gate.py

```diff
@@ -19,7 +19,8 @@
 from numpy import pi
 from ddt import ddt, data, unpack
 
-from qiskit import QuantumRegister, QuantumCircuit, QiskitError
+from qiskit import QuantumRegister, QuantumCircuit, QiskitError, transpile
+from qiskit.test import QiskitTestCase
```

Comment:
I don't think this line is needed. It duplicates with line 87.
In addition, could you please handle the black errors?